File: frontend/apps/ios/Pantopus/Core/Design/Components/BalanceHero.swift
//
//  BalanceHero.swift
//  Pantopus
//
//  Dark sky-gradient hero card for financial surfaces (Wallet,
//  Payments) showing an available balance, currency chip, and a
//  split-strip of secondary stats. Optional `holdTone` adds an
//  inline amber banner under the strip when payouts are paused.
//
//  Design reference: `docs/designs/A10/wallet-frames.jsx` (BalanceHero)
//  and `docs/new-design-parity.md` § A10.10.
//

import SwiftUI

/// Dark sky-gradient hero card surfacing an available balance.
///
/// The component is presentation-only — the caller supplies a
/// pre-formatted `amount` string (the leading "$" glyph is rendered
/// separately so it can be baseline-aligned at half scale), a
/// `currencyCode` for the glass chip, and 0–2 `SplitCell`s for the
/// glass split-strip below the amount.
///
/// Passing a `payoutFooter` swaps the card into the A14.6 Payments
/// shape: the currency chip + arcs + split-strip collapse and a
/// compact "Next payout · date" + frequency pill row renders below
/// the amount. The gradient and glass styling are preserved so both
/// surfaces read as the same primitive family.
public struct BalanceHero: View {
    /// Visual tone — `.holdTone` appends an inline amber banner under
    /// the split strip warning that payouts are paused.
    public enum Tone: String, Sendable, Hashable {
        case `default`
        case holdTone
    }

    /// A single cell in the glass split-strip.
    public struct SplitCell: Sendable, Hashable {
        public let icon: PantopusIcon?
        public let overline: String
        public let value: String
        public let note: String?

        public init(
            icon: PantopusIcon? = nil,
            overline: String,
            value: String,
            note: String? = nil
        ) {
            self.icon = icon
            self.overline = overline
            self.value = value
            self.note = note
        }
    }

    /// A14.6 Payments variant — replaces the currency chip + split
    /// strip with a compact two-element row under the amount:
    /// "Next payout · Mon, May 27" on the leading edge, a frequency
    /// glass pill ("Weekly") on the trailing edge.
    public struct PayoutFooter: Sendable, Hashable {
        public let nextPayoutLabel: String
        public let frequencyPill: String

        public init(nextPayoutLabel: String, frequencyPill: String) {
            self.nextPayoutLabel = nextPayoutLabel
            self.frequencyPill = frequencyPill
        }
    }

    private let overline: String
    private let amount: String
    private let currencyCode: String
    private let split: [SplitCell]
    private let tone: Tone
    private let holdHeadline: String?
    private let holdBody: String?
    private let payoutFooter: PayoutFooter?

    public init(
        overline: String,
        amount: String,
        currencyCode: String,
        split: [SplitCell] = [],
        tone: Tone = .default,
        holdHeadline: String? = nil,
        holdBody: String? = nil,
        payoutFooter: PayoutFooter? = nil
    ) {
        self.overline = overline
        self.amount = amount
        self.currencyCode = currencyCode
        self.split = split
        self.tone = tone
        self.holdHeadline = holdHeadline
        self.holdBody = holdBody
        self.payoutFooter = payoutFooter
    }

    public var body: some View {
        ZStack(alignment: .topTrailing) {
            if payoutFooter == nil {
                arcs
                    .padding(.top, -50)
                    .padding(.trailing, -40)
            }

            VStack(alignment: .leading, spacing: Spacing.s3) {
                topRow
                amountRow
                if let payoutFooter {
                    payoutFooterRow(payoutFooter)
                }
                if payoutFooter == nil, !split.isEmpty {
                    splitStrip
                }
                if tone == .holdTone {
                    holdBanner
                }
            }
            .padding(.horizontal, 18)
            .padding(.top, Spacing.s4)
            .padding(.bottom, 14)
        }
        .background(
            LinearGradient(
                stops: [
                    .init(color: Theme.Color.primary800, location: 0),
                    .init(color: Theme.Color.primary700, location: 0.55),
                    .init(color: Theme.Color.primary600, location: 1.0)
                ],
                startPoint: .topLeading,
                endPoint: .bottomTrailing
            )
        )
        .clipShape(RoundedRectangle(cornerRadius: 18, style: .continuous))
        .accessibilityElement(children: .combine)
        .accessibilityLabel(accessibility)
    }

    private var accessibility: String {
        if let payoutFooter {
            "\(overline): \(amount). \(payoutFooter.nextPayoutLabel), \(payoutFooter.frequencyPill)"
        } else {
            "\(overline): \(amount) \(currencyCode)"
        }
    }

    private var arcs: some View {
        ZStack {
            ForEach([90.0, 60.0, 30.0], id: \.self) { radius in
                Circle()
                    .stroke(Color.white, lineWidth: 1)
                    .frame(width: radius * 2, height: radius * 2)
            }
        }
        .opacity(0.18)
        .allowsHitTesting(false)
    }

    private var topRow: some View {
        HStack(alignment: .center) {
            Text(overline)
                .font(.system(size: 10.5, weight: .bold))
                .tracking(1.0)
                .textCase(.uppercase)
                .foregroundStyle(Theme.Color.primary200)
            Spacer(minLength: Spacing.s2)
            if payoutFooter == nil {
                currencyChip
            }
        }
    }

    private var currencyChip: some View {
        HStack(spacing: Spacing.s1) {
            Icon(.shieldCheck, size: 10, strokeWidth: 2.5, color: .white)
            Text(currencyCode)
                .font(.system(size: 10, weight: .bold))
                .tracking(0.4)
                .textCase(.uppercase)
                .foregroundStyle(.white)
        }
        .padding(.horizontal, Spacing.s2)
        .padding(.vertical, 3)
        .background(Color.white.opacity(0.16))
        .clipShape(Capsule())
    }

    private var amountRow: some View {
        let isCompact = payoutFooter != nil
        return HStack(alignment: .firstTextBaseline, spacing: Spacing.s1) {
            Text("$")
                .font(.system(size: isCompact ? 16 : 22, weight: .bold))
                .foregroundStyle(Theme.Color.primary200)
            Text(amount)
                .font(.system(size: isCompact ? 28 : 44, weight: isCompact ? .bold : .heavy))
                .tracking(isCompact ? -0.5 : -1.4)
                .foregroundStyle(.white)
                .monospacedDigit()
        }
    }

    private func payoutFooterRow(_ footer: PayoutFooter) -> some View {
        HStack(alignment: .center) {
            Text(footer.nextPayoutLabel)
                .pantopusTextStyle(.caption)
                .foregroundStyle(Color.white.opacity(0.85))
                .lineLimit(1)
            Spacer(minLength: Spacing.s2)
            Text(footer.frequencyPill)
                .font(.system(size: 10.5, weight: .bold))
                .tracking(0.4)
                .foregroundStyle(.white)
                .padding(.horizontal, 9)
                .padding(.vertical, 3)
                .background(Color.white.opacity(0.18))
                .clipShape(Capsule())
        }
    }

    private var splitStrip: some View {
        HStack(alignment: .top, spacing: Spacing.s0) {
            ForEach(Array(split.enumerated()), id: \.offset) { index, cell in
                splitCellView(cell)
                    .frame(maxWidth: .infinity, alignment: .leading)
                if index < split.count - 1 {
                    Rectangle()
                        .fill(Color.white.opacity(0.16))
                        .frame(width: 1)
                        .padding(.horizontal, Spacing.s3)
                }
            }
        }
        .padding(.horizontal, Spacing.s3)
        .padding(.vertical, 10)
        .background(Color.white.opacity(0.10))
        .overlay(
            RoundedRectangle(cornerRadius: Radii.lg, style: .continuous)
                .stroke(Color.white.opacity(0.14), lineWidth: 1)
        )
        .clipShape(RoundedRectangle(cornerRadius: Radii.lg, style: .continuous))
    }

    private func splitCellView(_ cell: SplitCell) -> some View {
        VStack(alignment: .leading, spacing: 2) {
            HStack(spacing: Spacing.s1) {
                if let icon = cell.icon {
                    Icon(icon, size: 10, strokeWidth: 2.5, color: Theme.Color.primary200)
                        .opacity(0.85)
                }
                Text(cell.overline)
                    .font(.system(size: 9.5, weight: .bold))
                    .tracking(0.6)
                    .textCase(.uppercase)
                    .foregroundStyle(Theme.Color.primary200.opacity(0.85))
            }
            Text(cell.value)
                .font(.system(size: 16, weight: .bold))
                .tracking(-0.25)
                .foregroundStyle(.white)
                .monospacedDigit()
            if let note = cell.note {
                Text(note)
                    .font(.system(size: 10.5))
                    .foregroundStyle(Theme.Color.primary200.opacity(0.8))
                    .lineLimit(1)
            }
        }
    }

    private var holdBanner: some View {
        HStack(alignment: .top, spacing: Spacing.s2) {
            Icon(.alertTriangle, size: 14, strokeWidth: 2.4, color: Theme.Color.warningLight)
            VStack(alignment: .leading, spacing: 1) {
                if let headline = holdHeadline {
                    Text(headline)
                        .font(.system(size: 11.5, weight: .bold))
                        .tracking(-0.05)
                        .foregroundStyle(Theme.Color.warningBg)
                }
                if let body = holdBody {
                    Text(body)
                        .font(.system(size: 10.5))
                        .foregroundStyle(Theme.Color.warningLight.opacity(0.9))
                }
            }
            Spacer(minLength: Spacing.s0)
        }
        .padding(.horizontal, 10)
        .padding(.vertical, Spacing.s2)
        .background(Theme.Color.warningLight.opacity(0.18))
        .overlay(
            RoundedRectangle(cornerRadius: 10, style: .continuous)
                .stroke(Theme.Color.warningLight.opacity(0.45), lineWidth: 1)
        )
        .clipShape(RoundedRectangle(cornerRadius: 10, style: .continuous))
    }
}

#Preview("BalanceHero variants") {
    VStack(spacing: Spacing.s4) {
        BalanceHero(
            overline: "Available to withdraw",
            amount: "847.50",
            currencyCode: "USD",
            split: [
                .init(
                    icon: .clock,
                    overline: "Pending",
                    value: "$186.00",
                    note: "3 tasks · clears by Dec 4"
                ),
                .init(
                    icon: .trendingUp,
                    overline: "This month",
                    value: "$1,284.50",
                    note: "8 tasks · ▲22% vs Oct"
                )
            ]
        )

        BalanceHero(
            overline: "Available to withdraw",
            amount: "847.50",
            currencyCode: "USD",
            split: [
                .init(
                    icon: .clock,
                    overline: "Pending",
                    value: "$186.00",
                    note: "3 tasks · clears by Dec 4"
                ),
                .init(
                    icon: .trendingUp,
                    overline: "This month",
                    value: "$1,284.50",
                    note: "8 tasks · ▲22% vs Oct"
                )
            ],
            tone: .holdTone,
            holdHeadline: "Withdrawals paused",
            holdBody: "Re-verify your bank to release funds."
        )

        // A14.6 Payments — compact payout-footer variant.
        BalanceHero(
            overline: "Available to pay out",
            amount: "124.50",
            currencyCode: "USD",
            payoutFooter: BalanceHero.PayoutFooter(
                nextPayoutLabel: "Next payout · Mon, May 27",
                frequencyPill: "Weekly"
            )
        )
    }
    .padding(Spacing.s4)
    .background(Theme.Color.appBg)
}

File: frontend/apps/android/app/src/main/java/app/pantopus/android/ui/components/BalanceHero.kt
@file:Suppress("MagicNumber", "LongMethod", "LongParameterList", "FunctionNaming", "UnusedPrivateMember")

package app.pantopus.android.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.offset
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.drawBehind
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import app.pantopus.android.ui.theme.PantopusColors
import app.pantopus.android.ui.theme.PantopusIcon
import app.pantopus.android.ui.theme.PantopusIconImage
import app.pantopus.android.ui.theme.Radii
import app.pantopus.android.ui.theme.Spacing

/**
 * Visual tone for [BalanceHero]. [HoldTone] appends an inline amber
 * banner under the split strip warning that payouts are paused.
 */
enum class BalanceHeroTone {
    Default,
    HoldTone,
}

/**
 * A single cell inside the glass split-strip below the amount.
 *
 * @property icon Optional leading glyph (e.g. clock for "Pending").
 * @property overline Uppercase mono caption (e.g. "PENDING").
 * @property value Headline number (e.g. "$186.00").
 * @property note Optional subline (e.g. "3 tasks · clears by Dec 4").
 */
data class BalanceHeroSplitCell(
    val overline: String,
    val value: String,
    val icon: PantopusIcon? = null,
    val note: String? = null,
)

/**
 * A14.6 Payments compact variant — replaces the currency chip + split
 * strip with a two-element row under the amount: "Next payout · date"
 * on the leading edge, frequency glass pill on the trailing edge.
 */
data class BalanceHeroPayoutFooter(
    val nextPayoutLabel: String,
    val frequencyPill: String,
)

/**
 * Dark sky-gradient hero card for financial surfaces (Wallet,
 * Payments). The Compose mirror of iOS
 * `Core/Design/Components/BalanceHero.swift` — same parameter names so
 * call sites read identically across the two platforms.
 *
 * @param overline Section label rendered in `primary200` mono caps.
 * @param amount Pre-formatted numeric balance — the `$` glyph is drawn
 *     separately at half size so it baseline-aligns under the amount.
 * @param currencyCode Three-letter code shown in the glass chip.
 * @param split 0–2 cells rendered in a glass split-strip below the
 *     amount. Cells beyond the first are separated by a vertical
 *     divider.
 * @param tone [BalanceHeroTone.HoldTone] appends an amber inline
 *     banner using [holdHeadline] / [holdBody].
 * @param holdHeadline Banner headline; only rendered when
 *     [tone] = [BalanceHeroTone.HoldTone].
 * @param holdBody Banner body copy; same gating as [holdHeadline].
 */
@Composable
fun BalanceHero(
    overline: String,
    amount: String,
    currencyCode: String,
    modifier: Modifier = Modifier,
    split: List<BalanceHeroSplitCell> = emptyList(),
    tone: BalanceHeroTone = BalanceHeroTone.Default,
    holdHeadline: String? = null,
    holdBody: String? = null,
    payoutFooter: BalanceHeroPayoutFooter? = null,
) {
    val gradient =
        Brush.linearGradient(
            colorStops =
                arrayOf(
                    0f to PantopusColors.primary800,
                    0.55f to PantopusColors.primary700,
                    1f to PantopusColors.primary600,
                ),
        )

    val accessibility =
        if (payoutFooter != null) {
            "$overline: $amount. ${payoutFooter.nextPayoutLabel}, ${payoutFooter.frequencyPill}"
        } else {
            "$overline: $amount $currencyCode"
        }

    Box(
        modifier =
            modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(18.dp))
                .background(gradient)
                .semantics { contentDescription = accessibility },
    ) {
        if (payoutFooter == null) {
            BalanceHeroArcs(
                modifier =
                    Modifier
                        .align(Alignment.TopEnd)
                        .offset(x = 40.dp, y = (-50).dp),
            )
        }
        Column(
            modifier =
                Modifier
                    .padding(horizontal = 18.dp, vertical = Spacing.s3)
                    .padding(top = Spacing.s1, bottom = 2.dp),
            verticalArrangement = Arrangement.spacedBy(Spacing.s3),
        ) {
            BalanceHeroTopRow(
                overline = overline,
                currencyCode = currencyCode,
                hideCurrencyChip = payoutFooter != null,
            )
            BalanceHeroAmount(amount = amount, compact = payoutFooter != null)
            if (payoutFooter != null) {
                BalanceHeroPayoutFooterRow(footer = payoutFooter)
            } else if (split.isNotEmpty()) {
                BalanceHeroSplitStrip(cells = split)
            }
            if (tone == BalanceHeroTone.HoldTone) {
                BalanceHeroHoldBanner(headline = holdHeadline, body = holdBody)
            }
        }
    }
}

@Composable
private fun BalanceHeroArcs(modifier: Modifier = Modifier) {
    Box(
        modifier =
            modifier
                .size(200.dp)
                .drawBehind {
                    listOf(90.dp.toPx(), 60.dp.toPx(), 30.dp.toPx()).forEach { radius ->
                        val centre = Offset(size.width / 2f, size.height / 2f)
                        drawCircle(
                            color = Color.White.copy(alpha = 0.18f),
                            radius = radius,
                            center = centre,
                            style = Stroke(width = 1f),
                        )
                    }
                },
    )
}

@Composable
private fun BalanceHeroTopRow(
    overline: String,
    currencyCode: String,
    hideCurrencyChip: Boolean = false,
) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Text(
            text = overline.uppercase(),
            color = PantopusColors.primary200,
            fontSize = 10.5.sp,
            fontWeight = FontWeight.Bold,
            letterSpacing = 1.sp,
        )
        Spacer(modifier = Modifier.weight(1f))
        if (!hideCurrencyChip) {
            Row(
                modifier =
                    Modifier
                        .clip(CircleShape)
                        .background(Color.White.copy(alpha = 0.16f))
                        .padding(horizontal = Spacing.s2, vertical = 3.dp),
                horizontalArrangement = Arrangement.spacedBy(Spacing.s1),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                PantopusIconImage(
                    icon = PantopusIcon.ShieldCheck,
                    contentDescription = null,
                    size = 10.dp,
                    strokeWidth = 2.5f,
                    tint = Color.White,
                )
                Text(
                    text = currencyCode.uppercase(),
                    color = Color.White,
                    fontSize = 10.sp,
                    fontWeight = FontWeight.Bold,
                    letterSpacing = 0.4.sp,
                )
            }
        }
    }
}

@Composable
private fun BalanceHeroAmount(
    amount: String,
    compact: Boolean = false,
) {
    Row(
        verticalAlignment = Alignment.Bottom,
        horizontalArrangement = Arrangement.spacedBy(Spacing.s1),
    ) {
        Text(
            text = "$",
            color = PantopusColors.primary200,
            fontSize = if (compact) 16.sp else 22.sp,
            fontWeight = FontWeight.Bold,
            modifier = Modifier.padding(bottom = if (compact) 4.dp else 8.dp),
        )
        Text(
            text = amount,
            color = Color.White,
            fontSize = if (compact) 28.sp else 44.sp,
            fontWeight = if (compact) FontWeight.Bold else FontWeight.ExtraBold,
            letterSpacing = if (compact) (-0.5).sp else (-1.4).sp,
        )
    }
}

@Composable
private fun BalanceHeroPayoutFooterRow(footer: BalanceHeroPayoutFooter) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Text(
            text = footer.nextPayoutLabel,
            color = Color.White.copy(alpha = 0.85f),
            fontSize = 12.sp,
            maxLines = 1,
        )
        Spacer(modifier = Modifier.weight(1f))
        Text(
            text = footer.frequencyPill,
            color = Color.White,
            fontSize = 10.5.sp,
            fontWeight = FontWeight.Bold,
            letterSpacing = 0.4.sp,
            modifier =
                Modifier
                    .clip(CircleShape)
                    .background(Color.White.copy(alpha = 0.18f))
                    .padding(horizontal = 9.dp, vertical = 3.dp),
        )
    }
}

@Composable
private fun BalanceHeroSplitStrip(cells: List<BalanceHeroSplitCell>) {
    Row(
        modifier =
            Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(Radii.lg))
                .background(Color.White.copy(alpha = 0.10f))
                .border(width = 1.dp, color = Color.White.copy(alpha = 0.14f), shape = RoundedCornerShape(Radii.lg))
                .padding(horizontal = Spacing.s3, vertical = 10.dp),
        verticalAlignment = Alignment.Top,
    ) {
        cells.forEachIndexed { index, cell ->
            BalanceHeroSplitCellView(
                cell = cell,
                modifier = Modifier.weight(1f),
            )
            if (index < cells.size - 1) {
                Box(
                    modifier =
                        Modifier
                            .width(1.dp)
                            .padding(horizontal = Spacing.s3)
                            .background(Color.White.copy(alpha = 0.16f)),
                )
            }
        }
    }
}

@Composable
private fun BalanceHeroSplitCellView(
    cell: BalanceHeroSplitCell,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier,
        verticalArrangement = Arrangement.spacedBy(2.dp),
    ) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(Spacing.s1),
        ) {
            cell.icon?.let { icon ->
                PantopusIconImage(
                    icon = icon,
                    contentDescription = null,
                    size = 10.dp,
                    strokeWidth = 2.5f,
                    tint = PantopusColors.primary200.copy(alpha = 0.85f),
                )
            }
            Text(
                text = cell.overline.uppercase(),
                color = PantopusColors.primary200.copy(alpha = 0.85f),
                fontSize = 9.5.sp,
                fontWeight = FontWeight.Bold,
                letterSpacing = 0.6.sp,
            )
        }
        Text(
            text = cell.value,
            color = Color.White,
            fontSize = 16.sp,
            fontWeight = FontWeight.Bold,
            letterSpacing = (-0.25).sp,
        )
        cell.note?.let { note ->
            Text(
                text = note,
                color = PantopusColors.primary200.copy(alpha = 0.8f),
                fontSize = 10.5.sp,
                maxLines = 1,
            )
        }
    }
}

@Composable
private fun BalanceHeroHoldBanner(
    headline: String?,
    body: String?,
) {
    Row(
        modifier =
            Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(10.dp))
                .background(PantopusColors.warningLight.copy(alpha = 0.18f))
                .border(width = 1.dp, color = PantopusColors.warningLight.copy(alpha = 0.45f), shape = RoundedCornerShape(10.dp))
                .padding(horizontal = 10.dp, vertical = Spacing.s2),
        horizontalArrangement = Arrangement.spacedBy(Spacing.s2),
        verticalAlignment = Alignment.Top,
    ) {
        PantopusIconImage(
            icon = PantopusIcon.AlertTriangle,
            contentDescription = null,
            size = 14.dp,
            strokeWidth = 2.4f,
            tint = PantopusColors.warningLight,
        )
        Column(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(1.dp),
        ) {
            headline?.let {
                Text(
                    text = it,
                    color = PantopusColors.warningBg,
                    fontSize = 11.5.sp,
                    fontWeight = FontWeight.Bold,
                    letterSpacing = (-0.05).sp,
                )
            }
            body?.let {
                Text(
                    text = it,
                    color = PantopusColors.warningLight.copy(alpha = 0.9f),
                    fontSize = 10.5.sp,
                )
            }
        }
    }
}

@Preview(showBackground = true, widthDp = 360, heightDp = 700, backgroundColor = 0xFFF6F7F9)
@Composable
private fun BalanceHeroPreview() {
    Column(
        modifier =
            Modifier
                .fillMaxWidth()
                .padding(Spacing.s4),
        verticalArrangement = Arrangement.spacedBy(Spacing.s4),
    ) {
        BalanceHero(
            overline = "Available to withdraw",
            amount = "847.50",
            currencyCode = "USD",
            split =
                listOf(
                    BalanceHeroSplitCell(
                        icon = PantopusIcon.Clock,
                        overline = "Pending",
                        value = "$186.00",
                        note = "3 tasks · clears by Dec 4",
                    ),
                    BalanceHeroSplitCell(
                        icon = PantopusIcon.TrendingUp,
                        overline = "This month",
                        value = "$1,284.50",
                        note = "8 tasks · ▲22% vs Oct",
                    ),
                ),
        )
        BalanceHero(
            overline = "Available to withdraw",
            amount = "847.50",
            currencyCode = "USD",
            split =
                listOf(
                    BalanceHeroSplitCell(
                        icon = PantopusIcon.Clock,
                        overline = "Pending",
                        value = "$186.00",
                        note = "3 tasks · clears by Dec 4",
                    ),
                ),
            tone = BalanceHeroTone.HoldTone,
            holdHeadline = "Withdrawals paused",
            holdBody = "Re-verify your bank to release funds.",
        )
        // A14.6 Payments — compact payout-footer variant.
        BalanceHero(
            overline = "Available to pay out",
            amount = "124.50",
            currencyCode = "USD",
            payoutFooter =
                BalanceHeroPayoutFooter(
                    nextPayoutLabel = "Next payout · Mon, May 27",
                    frequencyPill = "Weekly",
                ),
        )
        // padding hint
        Text(
            text = "Padding sanity hint",
            color = PantopusColors.appTextSecondary,
            fontSize = 10.sp,
            textAlign = TextAlign.Center,
            modifier = Modifier.fillMaxWidth().padding(PaddingValues(0.dp)),
        )
    }
}
